api/pom.xml

```diff
@@ -2,7 +2,7 @@
    <parent>
       <groupId>org.openmrs</groupId>
       <artifactId>openmrs</artifactId>
-      <version>1.12.0.SPRING-SNAPSHOT</version>
+      <version>1.12.0-SPRING4-SNAPSHOT</version>
```

Comment:
Might I suggest, just for consideration, that some change be made in the
Spring 4 version # even if it is to 1.12.1-SNAPSHOT. The reason is, after
the Spring 4 merge, what if someone wants to build an older Spring 3
version of 1.12.0-SNAPSHOT, perhaps to compare the behavior between the
two. If you're doing a full maven reactor build each time then everything
is fine. But say your .m2 has a Spring 4 version of 1.12.0-SNAPSHOT
core-api installed locally from a prior build, and then you switch to
Spring 3 and run a test in Spring 3 core-web. It will attempt to run
against the Spring 4 core-api jar.

It may sound unlikely, but this isn't just theoretical - I actually ran
into this issue. That is why I decided to re-version in the first place.
Then all of the jars that are built link together nicely and it's possible
to have both Spring 3 and Spring 4 versions of the app locally with no
conflicts.




On Mon, Mar 2, 2015 at 10:41 AM, Rafal Korytkowski <notifications@github.com
> wrote:

> In api/pom.xml
> <https://github.com/openmrs/openmrs-core/pull/1383#discussion_r25607364>:
>
> > @@ -2,7 +2,7 @@
> >     <parent>
> >        <groupId>org.openmrs</groupId>
> >        <artifactId>openmrs</artifactId>
> > -      <version>1.12.0.SPRING-SNAPSHOT</version>
> > +      <version>1.12.0-SPRING4-SNAPSHOT</version>
>
> It's needed for testing of the spring-4 branch. It will be back
> 1.12.0-SNAPSHOT once we merge with master.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openmrs/openmrs-core/pull/1383/files#r25607364>.
>